ideascube/library/migrations/0008_library_stock.py

```diff
@@ -26,6 +30,13 @@ def migrate_data(apps, schema_editor):
         new_book.save()
 
         for old_bookspecimen in old_book.specimens.all():
+            if old_bookspecimen.serial in specimens:
```

Comment:
Hum, not sure :)
Getting almost all the specimen, generating a dict and do a __contains__ on this dict may be longer than a request per old_bookspecimen.

It depends mainly of the number of specimens, old_books and old_books.specimens.
It's also depends on a lots of code optimization I don't know about :)

And premature optimization is the root of all evil.;)
Especially if this migration script will be run only once.

However, code is written then no need to rewrite it.